### PR TITLE
Accept #TMSH-VERSION in TMOS running config validation

### PR DESCRIPTION
**Pull request by ddog-nasirthomas**

### CODE REVIEW

File: pkg/networkconfigmanagement/profile/default_profiles.go [hunk 1]
```diff
@@ -330,7 +330,7 @@ var DefaultProfiles = Map{
 	"tmos": {
 		Name: "tmos",
 		Commands: CommandSet{
-			GetRunning: MkCommand("cat /config/partitions/*/bigip*.conf", `^sys global-settings\s*{`),
+			GetRunning: MkCommand("cat /config/partitions/*/bigip*.conf", `\A(#TMSH-VERSION:|sys global-settings\s*{)`),
```
Comment (garrison-stauffer) [inline pkg/networkconfigmanagement/profile/default_profiles.go]: Would it be possible to update this regex to include one more pattern for `ltm` configuration?

```
(^sys global-settings\s*{)|(^ltm (node|pool|virtual) \S+ {)|(^#TMSH-VERSION: \S+)
```

I think the `\A` might need to be removed too, I added some examples for what it'd hit here https://regex101.com/r/04Ct0e/1/list-substitution

File: pkg/networkconfigmanagement/profile/default_profiles_test.go [hunk 2]
```diff
@@ -115,6 +115,16 @@ func Test_DefaultProfiles_Running(t *testing.T) {
...
 }
 
+func Test_TMOSGetRunningValidator(t *testing.T) {
+	v := DefaultProfile(t, "tmos").Commands.GetRunning.Validator
+	assert.NoError(t, v.Validate("#TMSH-VERSION: 17.1.3\n"))
+	assert.NoError(t, v.Validate("sys global-settings {\n"))
+	assert.NoError(t, v.Validate("ltm virtual /Common/x {\n"))
+	assert.NoError(t, v.Validate("ltm pool /Common/pool_1 {\n"))
+	assert.NoError(t, v.Validate("ltm node /Common/node_1 {\n"))
+	assert.Error(t, v.Validate("not a tmos config header\n"))
```
Comment (garrison-stauffer) [inline pkg/networkconfigmanagement/profile/default_profiles_test.go:125]: not blocking - there are some fixtures folders Dan set up that we could add to for these specific cases. We might be able to add some new folders for other test cases, and do something like 
```
{
	name:                      "TMOS TMSH Header",
	profile:                   DefaultProfile(t, "tmos"),
	fixture:                   loadFixture("tmos", "running/tmsh-header"),
	expectedExtractedMetadata: &ExtractedMetadata{},
},
{
	name:                      "TMOS ltm check",
	profile:                   DefaultProfile(t, "tmos"),
	fixture:                   loadFixture("tmos", "running/ltm"),
	expectedExtractedMetadata: &ExtractedMetadata{},
},
```

Comment (garrison-stauffer) [reply pkg/networkconfigmanagement/profile/default_profiles_test.go:125]: that said, I do like just being able to test some of these precise edge cases. Will defer to @zoedt / @dplepage-dd on this

**Review by garrison-stauffer — APPROVED** [review]

File: pkg/networkconfigmanagement/profile/default_profiles_test.go (hunk 2 above)
Comment (ddog-nasirthomas) [reply pkg/networkconfigmanagement/profile/default_profiles_test.go:125]: Yeah, I saw those folders and thought about adding more, but I wasn't sure because I was worried about adding too many fixtures for each test case. I can make modifications to the test cases if that's how we want to move forward for these scenarios.

**Review by garrison-stauffer — APPROVED** [review]
